sklearn/ensemble/gradient_boosting.py

```diff
@@ -1866,3 +1872,310 @@ def apply(self, X):
         leaves = super(GradientBoostingRegressor, self).apply(X)
         leaves = leaves.reshape(X.shape[0], self.estimators_.shape[0])
         return leaves
+
+
+class GradientBoostingClassifierCV(GradientBoostingClassifier):
+    """Gradient Boosting for classification with CV to find best n_estimators
+
+    Gradient Boosting builds an additive model in a forward stage-wise fashion;
+    It allows for the optimization of arbitrary differentiable loss functions.
+    In each stage ``n_classes_`` regression trees are fit on the negative
+    gradient of the binomial or multinomial deviance loss function. Binary
+    classification is a special case where only a single regression tree is
+    induced.
+
+    This class allows automatically selecting the best number of boosting
+    stages (``n_estimators``) based on the cross-validated score computed
+    for the model after each additional boosting stage. In contrast to grid
+    search, which will retrain all the previous stages for each possible value
+    of ``n_estimators``, this class continues upon the previous model (with a
+    lower number of boosting stage) to save computational time.
+
+    Read more in the :ref:`User Guide <gradient_boosting>`.
+
+    .. versionadded:: 0.19
+
+    Parameters
+    ----------
+    loss : {'deviance', 'exponential'}, optional (default='deviance')
+        loss function to be optimized. 'deviance' refers to
+        deviance (= logistic regression) for classification
+        with probabilistic outputs. For loss 'exponential' gradient
+        boosting recovers the AdaBoost algorithm.
+
+    learning_rate : float, optional (default=0.1)
+        learning rate shrinks the contribution of each tree by `learning_rate`.
+        There is a trade-off between learning_rate and n_estimators.
+
+    cv_n_estimators : int or array-like of shape (n_cv_stages), (default=100)
```

Comment:
Do we use this `cv_` prefix elsewhere for `*CV` objects? We use `Cs`, `alphas`, etc. That convention is hard to adopt here. `learning_curve` uses `param_range`, and I think `n_estimators_range` would be okay here.